style/js/README.md

```diff
@@ -1,3 +1,190 @@
+JavaScript Style Guides
+=====
+
+Guides for programing JavaScript in style
+
+General
+---
+* Use [JSHint](http://jshint.com/)
+* Always use 'use strict'
+
+Parens and Braces
+---
+* Leave one space between 'if' and '('.
+
+```javascript
+if (something) {
+  //some work here;
+}
+```
+
+* Prefer this form for multiline if statments
+
+```javascript
+if (something) {
+  // do stuff
+  // even more stuff
+} else {
+  // some other stuff
+  // real stuff
+}
+```
+
+* On if statments, whenever it makes sense, you can omit the '{}'
+
+```javascript
+if (something)
+  return stuff;
+else
+  return otherStuff;
+```
+
+* Don't include spaces between 'function' and '('.
+
+```javascript
+function bar() {
+  return  foo;
+}
+
+var bar = function() {
+  return foo;
+}
+```
+
+Assignments and Declarations
+---
+* Initialize arrays with '[ ]'
+
+```javascript
+var array = [];
+```
+
+* Initialize objects with '{ }'
+
+```javascript
+var object = {};
+```
+
+* Leave one space after '{' and before '}' when initializing objects on the same line
+
+```javascript
+var map = { ready: 9, when: 4, "you are": 15 };
+```
+
+* Prefer this form for multiline object/array initialization
+
+```javascript
+var map = {
+  when: 4,
+  ready: 9,
+  "you are": 15
+};
+
+var map = [
+  1,
+  2,
+  3
+]
+```
+
+Constructors
+---
+
+```javascript
+function Foo(bar) {
+  this.bar = bar;
+}
+
+Foo.prototype.getBar = function() {
+  return this.bar;
+};
+
+var foo = new Foo("bar");
+
+// or
+
+function Foo(bar) {
+  this.bar = bar;
+
+  this.getBar = function() {
+    return this.bar;
+  }
+}
+
+var foo = new Foo("bar");
+```
+
+Chaining
+---
+
+* When chaining methods indent one time.
+
+```javascript
+var elements = [$('.button'), $('.navbar')];
+
+elements
+  .addClass("foo")
+  .children()
+  .html()
+  .end()
+  .appendTo("body");
+```
+
+Callbacks
+---
+
+* Whenever possible make callbacks named functions, it makes the stack trace easier to understand
+
+```javascript
+  function callback(error, result) {
+    //some work here
+  }
+
+  bar(foo, callback);
+```
+
+Conditional evaluation
+---
+
+* Always use `===`.
```

Comment:
@zamith on one client project and on theshelf-angular. I believe @azevedo-252 and @ronaldofs agree with me, that we will try to move into it as soon as possible.